src/ContainerAppHelper.ts

```diff
@@ -0,0 +1,435 @@
+        githubActionsToolHelper.debug(`Attempting to determine if Container App with name "${containerAppName}" exists in resource group "${resourceGroup}"`);
+        try {
+            let command = `containerapp show -n ${containerAppName} -g ${resourceGroup} -o none`;
+            let executionResult = await util.executeAndThrowIfError(`az`, command.split(' '));
+            return executionResult.exitCode === 0;
+        } catch (err) {
+            githubActionsToolHelper.warning(err.message);
+            return false;
+        }
+    }
+
+    /**
+     * Determines if the provided Container App Environment exists in the provided resource group.
+     * @param containerAppEnvironment - the name of the Container App Environment
+     * @param resourceGroup - the resource group that the Container App Environment is found in
+     * @returns true if the Container App Environment exists, false otherwise
+     */
+    public async doesContainerAppEnvironmentExist(containerAppEnvironment: string, resourceGroup: string): Promise<boolean> {
+        githubActionsToolHelper.debug(`Attempting to determine if Container App Environment with name "${containerAppEnvironment}" exists in resource group "${resourceGroup}"`);
+        try {
+            let command = `containerapp env show -n ${containerAppEnvironment} -g ${resourceGroup} -o none`;
+            let executionResult = await util.executeAndThrowIfError(`az`, command.split(' '));
+            return executionResult.exitCode === 0;
+        } catch (err) {
+            githubActionsToolHelper.warning(err.message);
+            return false;
+        }
+    }
+
+    /**
+     * Determines if the provided resource group exists.
+     * @param resourceGroup - the name of the resource group
+     * @returns true if the resource group exists, false otherwise
+     */
+    public async doesResourceGroupExist(resourceGroup: string): Promise<boolean> {
+        githubActionsToolHelper.debug(`Attempting to determine if resource group "${resourceGroup}" exists`);
+        try {
+            let command = `group show -n ${resourceGroup} -o none`;
+            let executionResult = await util.executeAndThrowIfError(`az`, command.split(' '));
+            return executionResult.exitCode === 0;
+        } catch (err) {
+            githubActionsToolHelper.warning(err.message);
+            return false;
+        }
+    }
+
+    /**
+     * Gets the default location for the Container App provider.
+     * @returns the default location if found, otherwise 'eastus2'
+     */
+    public async getDefaultContainerAppLocation(): Promise<string> {
+        githubActionsToolHelper.debug(`Attempting to get the default location for the Container App service for the subscription.`);
+        try {
+            let args = [`provider`, `show`, `-n`, `Microsoft.App`, `--query`, `resourceTypes[?resourceType=='containerApps'].locations[] | [0]`];
```

Comment:
I think we need to re-evaluate how we're constructing the command arguments that we provide to our execute function -- in this file I'm seeing some arguments that include piped values (`|`) and arguments that should be explicitly wrapped in double quotes, so I think we should chat again about the differences between what the command constructed from ADO's `tl.execSync` vs. the `core.exec` we use for GitHub Actions.

Reply:
Updated.